Index: src/c++/common/spds/SyncManager.cpp
===================================================================
RCS file: /cvsroot/sync4j/3x/client-api/native/src/c++/common/spds/SyncManager.cpp,v
retrieving revision 1.125
diff -c -r1.125 SyncManager.cpp
*** src/c++/common/spds/SyncManager.cpp	30 Jun 2008 14:16:56 -0000	1.125
--- src/c++/common/spds/SyncManager.cpp	14 Jul 2008 19:10:16 -0000
***************
*** 2002,2014 ****
      const char* itemName;
      Status *status = 0;
  
!     Source* s = item->getSource();
!     if (s) {
!         itemName = s->getLocURI();
      }
      else {
!         Target* t = item->getTarget();
!         itemName = t->getLocURI();
      }
  
      // Fill item -------------------------------------------------
--- 2002,2022 ----
      const char* itemName;
      Status *status = 0;
  
!     // During an "Add" we want the source URI, otherwise
!     // ("Delete", "Update") the target URI.
!     Target* t = item->getTarget();
!     if (t && strcmp(cmdInfo.commandName, ADD)) {
!         itemName = t->getLocURI();
      }
      else {
!         Source* s = item->getSource();
!         itemName = s->getLocURI();
!     }
! 
!     if (!itemName) {
!         // invalid command: no URI
!         status = syncMLBuilder.prepareItemStatus(cmdInfo.commandName, "", cmdInfo.cmdRef, 412);
!         return status;
      }
  
      // Fill item -------------------------------------------------
